airflow/gcp/hooks/pubsub.py

```diff
@@ -77,47 +106,110 @@ def publish(self, project: str, topic: str, messages: List[Dict]) -> None:
+            the topic may be stored. If not present, then no constraints
+            are in effect.
+        :type message_storage_policy:
+            Union[Dict, google.cloud.pubsub_v1.types.MessageStoragePolicy]
+        :param kms_key_name: The resource name of the Cloud KMS CryptoKey
+            to be used to protect access to messages published on this topic.
+            The expected format is
+            ``projects/*/locations/*/keyRings/*/cryptoKeys/*``.
+        :type kms_key_name: str
+        :param retry: (Optional) A retry object used to retry requests.
+            If None is specified, requests will not be retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: (Optional) The amount of time, in seconds, to wait for the request
+            to complete. Note that if retry is specified, the timeout applies to each
+            individual attempt.
+        :type timeout: float
+        :param metadata: (Optional) Additional metadata that is provided to the method.
+        :type metadata: Sequence[Tuple[str, str]]]
         """
-        service = self.get_conn()
-        full_topic = _format_topic(project, topic)
+        assert project is not None
+        publisher = self.get_conn()
+        topic_path = PublisherClient.topic_path(project, topic)  # pylint: disable=no-member
+
+        # Add airflow-version label to the topic
+        labels = labels or {}
+        labels['airflow-version'] = 'v' + version.replace('.', '-').replace('+', '-')
+
+        self.log.info("Creating topic (path) %s", topic_path)
         try:
-            service.projects().topics().create(  # pylint: disable=no-member
-                name=full_topic, body={}).execute(num_retries=self.num_retries)
-        except HttpError as e:
-            # Status code 409 indicates that the topic already exists.
-            if str(e.resp['status']) == '409':
-                message = 'Topic already exists: {}'.format(full_topic)
-                self.log.warning(message)
-                if fail_if_exists:
-                    raise PubSubException(message)
-            else:
-                raise PubSubException(
-                    'Error creating topic {}'.format(full_topic), e)
-
-    def delete_topic(self, project: str, topic: str, fail_if_not_exists: bool = False) -> None:
+            # pylint: disable=no-member
+            publisher.create_topic(
+                name=topic_path,
+                labels=labels,
+                message_storage_policy=message_storage_policy,
+                kms_key_name=kms_key_name,
+                retry=retry,
+                timeout=timeout,
+                metadata=metadata,
+            )
+        except AlreadyExists:
+            message = 'Topic already exists: {}'.format(topic)
+            self.log.warning(message)
+            if fail_if_exists:
+                raise PubSubException(message)
+        except GoogleAPICallError as e:
```

Comment:
This part of code is not covered by unit test. Can you complete missing unit tests?

Reply:
I've added test.